backend/src/main/java/it/chalmers/gamma/controller/administrationSubControllers/AuthorityAdminController.java

```diff
@@ -0,0 +1,125 @@
+                                    PostService postService, AuthorityLevelService authorityLevelService,
+                                    MembershipService membershipService){
+        this.authorityService = authorityService;
+        this.fkitService = fkitService;
+        this.postService = postService;
+        this.authorityLevelService = authorityLevelService;
+        this.membershipService = membershipService;
+    }
+
+    @RequestMapping(method = RequestMethod.POST)
+    public ResponseEntity<String> addAuthority(@RequestBody AuthorizationRequest request){      // TODO CHECK IF EXISTS
+        Post post = postService.getPost(UUID.fromString(request.getPost()));
+        if(post == null){
+            throw new PostDoesNotExistResponse();
+        }
+        FKITGroup group = fkitService.getGroup(UUID.fromString(request.getGroup()));
+        if(group == null){
+            throw new GroupDoesNotExistResponse();
+        }
+        AuthorityLevel level = authorityLevelService.getAuthorityLevel(UUID.fromString(request.getAuthority()));
+        if(level == null){
+            throw new AuthorityLevelNotFoundResponse();
+        }
+        authorityService.setAuthorityLevel(group, post, level);
+        return new AuthorityAddedResponse();
+    }
+
+    @RequestMapping(value = "/user_authorities/{authorityLevel}", method = RequestMethod.GET)
+    public List<JSONObject> getUsersWithAuthority(@PathVariable("authorityLevel") String level){
+        System.out.println(level);
+        AuthorityLevel authorityLevel = authorityLevelService.getAuthorityLevel(UUID.fromString(level));
+        List<Authority> authorities = authorityService.getAllAuthoritiesWithAuthorityLevel(authorityLevel);
+        ITUserSerializer serializer = new ITUserSerializer(ITUserSerializer.Properties.getAllProperties());
+        List<JSONObject> users = new ArrayList<>();
+        for(Authority authority : authorities){
+            FKITGroup group = authority.getId().getFkitGroup();
+            Post post = authority.getId().getPost();
+            List<ITUser> userDatas = membershipService.getUserByGroupAndPost(group, post);
+            for(ITUser user : userDatas){
+                users.add(serializer.serialize(user, null));
+            }
+        }
+        return users;
+    }
+
+    @RequestMapping(value = "/{id}", method = RequestMethod.DELETE)
+    public ResponseEntity<String> removeAuthorization(@PathVariable("id") String id){
+        authorityService.removeAuthority(UUID.fromString(id));
+        return new AuthorityRemovedResponse();
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<List<Authority>> getAllAuthorities(){
+        List<Authority> authorities = authorityService.getAllAuthorities();
+        return new GetAllAuthoritiesResponse(authorities);
+    }
+
+    // BELOW THIS SHOULD MAYBE BE MOVED TO A DIFFERENT FILE
```

Comment:
// BELOW THIS SHOULD MAYBE BE MOVED TO A DIFFERENT FILE

Reply:
This is meant in a final build with more functions